synapse/federation/federation_server.py

```diff
@@ -611,7 +611,7 @@ def _transaction_from_pdus(self, pdu_list):
                 logger.info("Accepting join PDU %s from %s", pdu.event_id, origin)
 
         # We've already checked that we know the room version by this point
-        room_version = await self.store.get_room_version_id(pdu.room_id)
+        room_version = await self.store.get_room_version(pdu.room_id)
```

Comment:
One difference here is that `get_room_version_id` has an LRU cache in front of it while `get_room_version` does not. I'm unsure how much this matters in practice, but thought it was worth pointing out.

Reply:
It doesn't really matter, because `get_room_version` is a thin wrapper around `get_room_version_id`; the slow bit (going to the db) is still cached.